### DISCUSSION TITLE
Plus addressing for mailboxes

### DISCUSSION THREAD

**Discussion by YeapGuy** [simple-login/app#2005] [Feature Requests]
It'd be great if I could select a tag for an alias, and then all emails sent to that alias would be forwarded to its inbox **plus that selected tag**. Quick mockup I made:
![image](https://github.com/simple-login/app/assets/43749899/3ef3f637-240a-44b1-a4ce-a4088867c4a1)
So, if like in the image, I selected tag `newsletters`, I'd get the email from newsletter-sender@example.net at mailbox+newsletters@example.com instead of mailbox@example.com.
It doesn't have to be a dropdown, could just be a text field.
This would be super useful for me, because with my current workflow, I need to create **and verify** separate mailboxes for each of these tags.

**Comment by springcomp** [comment]
Directories do that automatically albeit in the other way around from what you describe.

Just create a new directory named `newsletter` and you can receive mails to `newsletter+mailbox@example.com`.
Basically, you can disclose addresses like `newsletter+anything@example.com` and the corresponding alias will be created on-the-fly.

![image](https://github.com/simple-login/app/assets/8488398/ddcf61cd-be83-47ef-a8e1-6474901dd158)

**Reply by YeapGuy** [reply]
Okay, I see what you mean, this could help, but what I'm requesting would still be very useful for many reasons (not having to create & verify separate mailboxes for each "tag", ability to tag any alias, not just directory-created ones...)

**Reply by luuvasara** [reply]
> but what I'm requesting would still be very useful for many reasons (not having to create & verify separate mailboxes for each "tag", ability to tag any alias, not just directory-created ones...)

I find it strange that simplelogin doesn't support this out of the box, it is after all basically the founding feature used by themselves
https://en.wikipedia.org/wiki/Email_address#Sub-addressing

**Reply by flagel** [reply]
Directories have their use but they also have limitations compared to doing the tagging on the backend with plus addresses. The sender can always tell that SimpleLogin is the active MTA for a recipient. 

Knowing that:
1) The sender can strip anything up to the last "/", "#", or "+" and still get through, bypassing any rules the recipient might have set up for those tags.
2) The sender can manipulate metadata used for backend processing. Instead of foobar/alias@domain.tld they can now send an invoice to billing/alias@domain.tld or amazon/alias@domain.tld in the hopes that gets paid in error.

Besides, not all senders accept "/", "#", or "+" as part of a mail address, in which case you end up with no tagging at all.

I understand that not all parties you will forward to will honour plus addressing, but you should make an exception for those that you know do (e.g. ProtonMail). Not great design.

**Comment by cartesiancat** [comment]
I tried adding a "new" mailbox with a plus-addressed alias, e.g. first.last+newsletters@gmail.com and first.last+travel@gmail.com, but SimpleLogin wouldn't let me. It would be so useful if it did because then I could allocate each SimpleLogin alias to a "tag" in my mail provider which would then send those messages into the correct folder at the very end.
